backend/de.metas.business/src/main/java/de/metas/bpartner/impexp/blockstatus/BPartnerBlockStatusImportProcess.java

```diff
@@ -0,0 +1,132 @@
+import de.metas.bpartner.blockstatus.BlockStatus;
+import de.metas.bpartner.blockstatus.CreateBPartnerBlockStatusRequest;
+import de.metas.bpartner.blockstatus.file.BPartnerBlockFileService;
+import de.metas.bpartner.service.IBPartnerDAO;
+import de.metas.impexp.processing.ImportRecordsSelection;
+import de.metas.impexp.processing.SimpleImportProcessTemplate;
+import de.metas.organization.OrgId;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.util.lang.IMutable;
+import org.compiere.SpringContextHolder;
+import org.compiere.model.I_C_BPartner;
+import org.compiere.model.I_C_BPartner_BlockStatus;
+import org.compiere.model.I_I_BPartner_BlockStatus;
+import org.compiere.model.X_I_BPartner_BlockStatus;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Properties;
+
+/**
+ * Imports {@link I_I_BPartner_BlockStatus} records to {@link org.compiere.model.I_C_BPartner_BlockStatus}.
+ **/
+public class BPartnerBlockStatusImportProcess extends SimpleImportProcessTemplate<I_I_BPartner_BlockStatus>
+{
+	private final IBPartnerDAO bPartnerDAO = Services.get(IBPartnerDAO.class);
+	private final BPartnerBlockStatusService bPartnerBlockStatusService = SpringContextHolder.instance.getBean(BPartnerBlockStatusService.class);
+	private final BPartnerBlockFileService bPartnerBlockFileService = SpringContextHolder.instance.getBean(BPartnerBlockFileService.class);
+
+	@Override
+	public Class<I_I_BPartner_BlockStatus> getImportModelClass()
+	{
+		return I_I_BPartner_BlockStatus.class;
+	}
+
+	@Override
+	public String getImportTableName()
+	{
+		return I_I_BPartner_BlockStatus.Table_Name;
+	}
+
+	@Override
+	protected String getTargetTableName()
+	{
+		return I_C_BPartner_BlockStatus.Table_Name;
+	}
+
+	@Override
+	protected void updateAndValidateImportRecords()
+	{
+		final ImportRecordsSelection selection = getImportRecordsSelection();
+
+		BPartnerBlockStatusImportTableSqlUpdater.updateAndValidateBPBlockStatusImportTable(selection);
+
+		final int countErrorRecords = BPartnerBlockStatusImportTableSqlUpdater.countRecordsWithErrors(selection);
+
+		getResultCollector().setCountImportRecordsWithValidationErrors(countErrorRecords);
+	}
+
+	@Override
+	protected ImportRecordResult importRecord(
+			final @NonNull IMutable<Object> stateHolder,
+			final @NonNull I_I_BPartner_BlockStatus importRecord,
+			final boolean isInsertOnly) throws Exception
+	{
+		bPartnerDAO.getBySAPBpartnerCode(importRecord.getSAP_BPartnerCode())
+				.stream()
+				.map(partner -> buildCreateBPBlockStatusRequest(importRecord, partner))
+				.forEach(bPartnerBlockStatusService::createBPartnerBlockStatus);
+
+		return ImportRecordResult.Inserted;
+	}
+
+	@Override
+	protected String getImportOrderBySql()
+	{
+		return I_I_BPartner_BlockStatus.COLUMNNAME_I_BPartner_BlockStatus_ID;
+	}
```

Comment:
Why not per partner?

Reply:
as `I_BPartner_BlockStatus_ID` is the best indicator we have for what the order of the rows was in the imported file... that way we have a predictable outcome.